src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -455,6 +460,11 @@ public void Abort(ConnectionAbortedException error)
                 return;
             }
 
+            if (reason != ConnectionEndReason.NoError && _metricsTagsFeature != null)
+            {
+                _metricsTagsFeature.TryAddTag(KestrelMetrics.KestrelConnectionEndReason, reason.ToString());
```

Comment:
The PR description claimed the end reason would always be present, but reason == ConnectionEndReason.NoError seems to be a case where it isn't. Is it a bug if reason==NoError?

Reply:
The PR code hasn't caught up with the description yet. I'm in the middle of changing it to always record the reason. `NoError` will disappear.

Reply:
Improved. Reason is always added. Still need to set `error.type`.